hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/CreateTableProcedure.java

```diff
@@ -114,6 +115,7 @@ protected Flow executeFromState(final MasterProcedureEnv env, final CreateTableS
           // XXX: this stage should be named as set table enabled, as now we will cache the
           // descriptor after writing fs layout.
           setEnabledState(env, getTableName());
+          env.getMasterServices().getTableStateManager().finishCreating(getTableName());
```

Comment:
Actually, the sub procedure will be execute after this line has been executed, so the solution here can not solve the problem described in the issue...